### ISSUE TITLE
[Bug]: Can't Open Privacy Policy

### ISSUE THREAD

**Issue by shashankiitbhu** [commons-app/apps-android-commons#5532]
### Summary

Unable to Open the Privacy Policy on the app's login page. This can become a hurdle in getting the app Review in Release as Play Store Guidelines requires there to be accessible privacy policy

### Steps to reproduce

1. Open the app
2. Logout if you are logged in
3. click on "Privacy Policy"

### Expected behaviour

Privacy Policy Opens in a Web View

### Actual behaviour

Misleading Error Message

### Device name

Xaomi 11 Lite NE

### Android version

13

### Commons app version

4.2.1-debug-main

### Screen-shots

https://github.com/commons-app/apps-android-commons/assets/126143257/75c80d7f-98ef-4450-82ba-57ab9fc2bc2b

### Would you like to work on the issue?

Yes

**Comment by kanahia1** [comment]
Hey @shashankiitbhu it is working fine for me 😀

https://github.com/commons-app/apps-android-commons/assets/114223204/4cf90791-a263-4a92-9ed3-3b3118976925



![Screenshot_20240209_143727_Settings](https://github.com/commons-app/apps-android-commons/assets/114223204/5974ddf3-3fe6-4e86-8562-880dd0789453)

**Comment by nicolas-raoul** [comment]
Thanks both for the feedback!

This link is important so it should be fixed to work on all devices.

@shashankiitbhu would you mind installing some intent interceptor tool and checking what intent is sent? Do you have any other app where such links work? If yes please post these intents too, in order to compare. Thanks a lot! :-)

**Comment by rohit9625** [comment]
> Hey @shashankiitbhu it is working fine for me 😀
> 
> https://github.com/commons-app/apps-android-commons/assets/114223204/4cf90791-a263-4a92-9ed3-3b3118976925
> 
> 
> 
> ![Screenshot_20240209_143727_Settings](https://github.com/commons-app/apps-android-commons/assets/114223204/5974ddf3-3fe6-4e86-8562-880dd0789453)
> 

Did you tried it on a latest branch?

**Comment by shashankiitbhu** [comment]
@rohit9625 Yes, I am looking into this issue - to find out if it's device specific

**Comment by shashankiitbhu** [comment]
@nicolas-raoul I Found out that this issue was due to the fact that I had Github Mobile App Installed which expects deep links like the link we are using for Privacy Policy but a conditional check in ```handleWebUrl``` Method was returning the method after checking if the link opens in a browser or not (Which is misleading for cases like this), for example, if we try to open Wikipedia link but Wikipedia App is already installed (assuming deep links are set up in Wikipedia app) then it would cause the same issue as this one

**Comment by ShashwatKedia** [comment]
> @nicolas-raoul I Found out that this issue was due to the fact that I had Github Mobile App Installed which expects deep links like the link we are using for Privacy Policy but a conditional check in `handleWebUrl` Method was returning the method after checking if the link opens in a browser or not (Which is misleading for cases like this), for example, if we try to open Wikipedia link but Wikipedia App is already installed (assuming deep links are set up in Wikipedia app) then it would cause the same issue as this one

Um, @shashankiitbhu, I also have the GitHub Mobile app installed on my device, but I am still able to open the privacy policy from my device, and it successfully opens in the GitHub mobile app itself.

**Comment by shashankiitbhu** [comment]
@ShashwatKedia That's Interesting, can you tell me more about your Android Version and API Level, Deep Links or App Links Behave differently on different API levels , I would love to investigate this more

**Comment by ShashwatKedia** [comment]
@shashankiitbhu Sure! My device is a OnePlus Nord CE 2 Lite, running API level 31. It's nice to see that you are so much interested in investigating deep into the issue :)

**Comment by shashankiitbhu** [comment]
@ShashwatKedia This Issue is tested well on a Device with API Level 33 (or Android version 13) and I have also found other Deep Link (App Link) Issues with this API Level (on other Projects). It seems to work fine with older API Levels. Hence, The Proposed solution is for devices running on API Level 33, Thanks :)

**Comment by shashankiitbhu** [comment]
https://github.com/commons-app/apps-android-commons/assets/126143257/a9446b94-a64c-408f-8dff-7cb80eb0b014

As you can see here the app directly stops the Privacy Policy Url from opening

**Comment by nicolas-raoul** [comment]
I merged for now, everyone who had the issue please try and let us know here, thanks! :-)
